code-challenges.js

```diff
@@ -1,64 +1,117 @@
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
-
+// THIS IS NOT WORKING I COULNDT FIGURE IT OUT :(
 
 
+// var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"];
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
 
+//     // turn an array into string
+// var randomNounsStr = randomNouns.join(', ');
+//     // give a string a function
 
-// --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
+// for (let i = 0; i < randomNounsStr.length; i++) {
+//     if (typeof randomNounsStr[i] === 'string') {
+//         randomNounsStr[i] = randomNounsStr[i].toUpperCase();
+//     }
+// console.log(randomNounsStr());
 
-var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
-// Expected output: [-8, 0, 8, 46, 59, 90, 107]
+// // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
 
+// var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"];
+// // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+// const onlyNums = (array) => {
+//     let newArr = [];
+//     for(let i=0; i<array.length; i++){
+//         newArr.push(array[i] > 1);
+//     }
+//     return newArr;
+// };
 
+// console.log(onlyNums(mixedDataArray));
 
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
 // Use the test variables provided.
 
-var vowelTester1 = "learn"
-// Expected output: 1
-var vowelTester1 = "throw"
-// Expected output: 3
-
-
-
-
-
-
-// --------------------5) Create a function that takes three arguments - two numbers and a mathematical operation (+, -, *, /) and performs the given calculation. If the input tries to divide by 0, return: "Can't divide by 0!"
-
-
-
+// var vowelTester1 = "learn";
+// // Expected output: 1
+// var vowelTester2 = "throw";
+// // Expected output: 3
+
+
+// const pickOne = (array) => {
+//     let newArr = array.split("").filter(value => {
+//         let checkIndex = value.toLowerCase();
+//         return checkIndex === "e" || checkIndex === "o";
+//     });
+//     return newArr;
+// };
+// console.log(pickOne(vowelTester1));
+// console.log(pickOne(vowelTester2));
+
+
+// --------------------5) Create a function that 
+// takes three arguments - 
+// two numbers 
+// and a mathematical operation (+, -, *, /) 
+// and performs the given calculation. 
+// If the input tries to divide by 0, 
+// return: "Can't divide by 0!"
+
+function calculator(x, y, z) {
+    if (y === "*") {
+        return (x) * (z);
+    } else if (y === "+") {
+        return (x) + (z);
+    }else if (y === "-") {
+        return (x) - (z);
+    }else if (z === 0) {
+        return "Can't divide by 0!"
+    }
+}
```

Comment:
this logic works! think about using different variables than x, y, z though so that our code is a little clearer